msupply-foundation/open-msupply
# 2857 suggest next encounter date

Fixes #2857 

# 👩🏻‍💻 What does this PR do? 
- Introduces a new EncounterNode property to fetch `suggested_next_encounter` information.
- When creating an encounter, autoselects the type of the latest encounter.
- Also sets the suggested next encounter date if provided
- Mark the suggested day in the date picker

![image](https://github.com/msupply-foundation/open-msupply/assets/88299195/095fbc41-c7cb-4ce3-9714-77d501de2bee)


# 🧪 How has/should this change been tested? 

Please use the latest schema from the schema repo to test this change.
- Create a patient with an HIVCareEncounter and some dispensed pills
- Check the end of supply date in this encounter
- Save as visited
- Create new encounter
- The modal should have HIV Care Encounter preselected
- Furthermore the end of supply date should be selected
- Open calendar and hover the marked day
- Suggested dates in the past should be ignored
